### DISCUSSION TITLE
pyAEDT on linux

### DISCUSSION THREAD

**Discussion by erikajob91** [ansys/pyaedt#5157] [Q&A] [answered]
Hi, 
is pyAEDT working on Linux ? it's not clear to me how to install it because the cpython in my ANSYS installation is only for windows. Here

https://aedt.docs.pyansys.com/version/stable/Getting_started/Installation.html

there are cited some env variables to be set, but what for ? Am I able to access my model in HFSS using the these api ? For example, if I need to steer the beam for an antenna array, will I able to write a script ? Unfortunately, in Linux the script for the steering is not working ?

**Comment by Samuelopez-ansys** [comment, marked as answer]
Hi @erikajob91 ,

PyAEDT is Linux compatible.

But you need at least AEDT 2023R1, because it is the first version that includes GRPC protocol which is the enabler of CPython in Linux. Before it was not possible because the communication protocol was COM, and it is Windows base.

I encourage you to use 2024R2, because by default AEDT opens a GRPC session. In previous versions, when the user opened AEDT, a COM session was opened.

Coming back to PyAEDT, if your script is going to open a new AEDT session, you need to set the 2 environment variables mentioned in the documentation, because the AEDT Installer does not create them. Once you set them(with the correct path) , you can activate your python interpreter and run your script. You can try a simple call like:

import pyaedt

App=pyaedt.Desktop()

Finally, you can use the pyaedt installer, but before launching AEDT, you need to set the env. Variables and launch AEDT in GRPC mode, as I said before 2024R2, it was not by default, and you needto add this argument 

ansysedt.exe -grpcsrv 50051

**Reply by erikajob91** [reply]
Hi,
thank you for the explanation.

**Reply by erikajob91** [reply]
Hi,
I requested the 2024R2, but meanwhile is there any way to open manually a GRPC session in the 2024R1 in order to point to my interpreter in the conda env I have just created ?

thank you

**Reply by erikajob91** [reply]
In my case the variables  should be se as follows ?

export ANSYSEM_ROOT241=/archive/apps/ansys_inc/v241-sp3/v241/Linux64
export LD_LIBRARY_PATH=$ANSYSEM_ROOT241/common/mono/Linux64/lib64:$ANSYSEM_ROOT241/Delcross:$LD_LIBRARY_PATH
----------------------

(base) [epellegrino@dvnode050 hfss-project]$ module show ansys/2024R1-SP3
-------------------------------------------------------------------
/cm/shared/modulefiles/ansys/2024R1-SP3:

module-whatis   {adds ANSYS CFX/Fluent, EM to your environment variables}
conflict        ansys
prepend-path    PATH /archive/apps/ansys_inc/v241-sp3/v241/CFX/bin
prepend-path    PATH /archive/apps/ansys_inc/v241-sp3/v241/fluent/bin
prepend-path    PATH /archive/apps/ansys_inc/v241-sp3/v241/TurboGrid/bin
prepend-path    PATH /archive/apps/ansys_inc/v241-sp3/v241/Framework/bin/Linux64
prepend-path    PATH /archive/apps/ansys_inc/v241-sp3/v241/ansys/bin
prepend-path    PATH /archive/apps/ansys_inc/v241-sp3/v241/ansys/bin/linx64
prepend-path    PATH /archive/apps/ansys_inc/v241-sp3/v241/Linux64
prepend-path    PATH /archive/apps/ansys_inc/v241-sp3/v241/optiSLang
prepend-path    PATH /archive/apps/ansys_inc/v241-sp3/v241/fensapice/bin
setenv          ANSYS_ROOT /archive/apps/ansys_inc
setenv          ANSYS_VERSION v241-sp3/v241
setenv          ANSYS241_DIR /archive/apps/ansys_inc/v241-sp3/v241/ansys
setenv          ANSYSLIC_DIR /archive/apps/ansys_inc/shared_files/licensing
setenv          ANSYSLMD_LICENSE_FILE 1055@10.87.0.77

**Comment by Samuelopez-ansys** [comment]
Yes, it should be like it.

Also specify in the pyaedt call the version:

pyaedt.Hfss(version="2024.1")
